signac/core/synceddict.py

```diff
@@ -101,16 +55,19 @@ def _validate_key(key):
         return key
 
     def _dfs_convert(self, root):
-        if type(root) is type(self):
+        if isinstance(root, _SyncedDict):
             for k in root:
                 root[k] = self._dfs_convert(root[k])
+        elif isinstance(root, _SyncedList):
+            for i, k in enumerate(root):
+                root[i] = self._dfs_convert(k)
         elif isinstance(root, Mapping):
-            ret = type(self)(parent=self)
+            ret = _SyncedDict(parent=self)
```

Comment:
I believe the reason that unit tests are failing is because the `_SyncedDict` is a mixed-in class. Other classes such as `JSONDict` inherit from this class. It is therefore important to return `type(self)(...)` and not `_SyncedDict(...)` here.

I am not sure whether the same holds for the `_SyncedList` block, we may need a unit test that explicitly tests inheritance.

Reply:
I tried swapping that out for `type(self)` but it didn't seem to fix the problem.

Reply:
Are you sure that this is triggered at all and if yes, can you check which types are triggering it?

Reply:
`type(root)` is `dict`, `type(self)` is `signac.core.attrdict.SyncedAttrDict`